app/src/main/java/com/topjohnwu/magisk/arch/BaseUIActivity.kt

```diff
@@ -59,6 +62,31 @@ abstract class BaseUIActivity<VM : BaseViewModel, Binding : ViewDataBinding> :
                 directionsDispatcher.value = null
             }
         }
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            window?.decorView?.let {
+                it.systemUiVisibility = (it.systemUiVisibility
+                        or View.SYSTEM_UI_FLAG_LAYOUT_STABLE
+                        or View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
+                        or View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION)
+            }
+
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                window?.decorView?.post {
+                    // If navigation bar is short enough (gesture navigation enabled), make it transparent
```

Comment:
How about using `resources.getIdentifier("config_navBarInteractionMode", "integer", "android")`?

Reply:
Maybe some OEMs have their own implementation?

Reply:
Okay then. Actually, I mentioned this because I found this https://github.com/aosp-mirror/platform_packages_apps_settings/blob/ffae58c97d4c84fab7aba1c55be03e4265a933f2/src/com/android/settings/gestures/SystemNavigationPreferenceController.java#L100-L103.